packages/instrument-utils/src/storedInstrument/AccountInstrumentSelect/AccountInstrumentSelect.tsx

```diff
@@ -289,7 +289,7 @@ class AccountInstrumentSelect extends PureComponent<AccountInstrumentSelectProps
         const { selectedInstrumentId } = this.props;
 
         if (prevSelectedInstrumentId !== selectedInstrumentId) {
-            this.updateFieldValue(selectedInstrumentId);
+            setTimeout(() => this.updateFieldValue(selectedInstrumentId));
```

Comment:
Just for clarification

I have faced with the case when `updateFieldValue` function calls from `componentDidUpdate` and than from `componentDidMount` and as a result field value is updated with outdated value. It happens because setTimeout calls in the end when stack is empty. To fix this i added `setTimeout` to `componentDidUpdate`

Reply:
Interesting seems like this is already what we did in `componentDidMount`, can you please add https://github.com/bigcommerce/checkout-js/pull/1323/files#diff-8eb58cc624cc62a268cc751632f75b94df5efcc4a7fc29c11be32f6a065d4825R281 comment here as well please.